File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let answer = 0;
  
    for (let i = 1; i <= number; i++) {
        let cnt = 0;
        for (let j = 1; j * j <= i; j++) {
            if (i % j === 0) {
                cnt += (j * j === i) ? 1 : 2;
            }
        }
        if(cnt > limit) cnt = power;
        answer += cnt
    }
  
    return answer;
}

File: Kotlin/프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.kt
class Solution {
    fun solution(number: Int, limit: Int, power: Int): Int {
        return IntArray(number) { getCount(it + 1) }.fold(0) { acc, v ->
            if (v > limit) acc + power
            else acc + v
        }
    }

    private fun getCount(n: Int): Int {
        var count = 0
        var i = 1
        while (i * i < n) {
            if (n % i++ == 0) count += 2
        }
        if (i * i == n) count++
        return count
    }
}
